### PR TITLE
RD-2360 deprecate numCategories, add singleLabel, scoreThreshold opti…

### PR DESCRIPTION
**Pull request by iredpath**
…ons for categories, add raw score to tcat response object

### CODE REVIEW

**Comment by iredpath** [comment]
we'll ultimately want version 1.10.104 here, so let's wait until #130 gets merged to master (and subsequently merged into this branch) before releasing

File: model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java
Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java]: Let's box this per our convention. If both this and `threshold` are set, who wins?

Comment (iredpath) [reply model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java]: `threshold` wins with respect to the threshold value itself, but we will still only return a maximum of one result if both are set

File: model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java
Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java:55]: maybe worthwhile to clarify that the threshold will be ignored if `singleLabel` is `true`, especially if the single label score falls below the threshold? or did I interpret it wrong?

Comment (iredpath) [reply model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java:55]: turning on `singleLabel` automatically sets `scoreThreshold` to -INF, but this will get overridden if `scoreThreshold` is also specified.  I'll add a clarifying note to `singleLabel` that the highest scoring category gets returned regardless of score (although after talking to Hannah we may hold off on exposing the raw score altogether, so this may all be moot)

Comment (guruxu) [reply model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java:55]: So not merging yet?

Comment (iredpath) [reply model/src/main/java/com/basistech/rosette/apimodel/CategoriesOptions.java:55]: got the okay from Hannah, so we're sticking with these changes.  Just waiting on the verification.

**Review by guruxu — APPROVED** [review]

**Pull request merged**
